Shared/Realm.Shared/RealmObject.cs

```diff
@@ -32,11 +37,58 @@ namespace Realms
     /// Has a Preserve attribute to attempt to preserve all subtypes without having to weave.
     /// </remarks>
     [Preserve(AllMembers = true, Conditional = false)]
-    public class RealmObject : IReflectableType
+    public class RealmObject : IReflectableType, INotifyPropertyChanged
     {
+        #region static
+
+        static RealmObject()
+        {
+            NativeCommon.register_notify_realm_object_changed(NotifyRealmObjectPropertyChanged);
```

Comment:
Yeah, I don't have strong feelings about where it should go - as you said, it's slightly more related to object, so I left it here, but I don't mind moving it to `Realm`.